addons/mail/static/src/models/message_view.js

```diff
@@ -119,6 +120,19 @@ registerModel({
                 this.threadViewOwnerAsLastMessageView.handleVisibleMessage(this.message);
             }
         },
+        /**
+         * This listens to the right click event, and used to redirect the event
+         * as a dialog.
+         *
+         * @param {Event} ev
+         */
+         async onContextMenu(ev) {
+            ev.stopPropagation();
+            ev.preventDefault();
```

Comment:
Don't `stopPropagation`, this is a bad design.
The few ones we have in discuss code are either to prevent invoking behaviour outside of discuss code, or we made mistakes and it's a bit costly to change them.

The better approach is to flag events with `markAsHandled` and have bubbling handlers have conditions on `isEventHandled`. This makes code more maintainable, and give more room to implement a specific behaviour that fully follows desired specs.

I suppose the `preventDefault` is to not let browser invoke the right-click menu from the browser/OS, right?
If so, then totally ok to keep this one.

Reply:
I do not have other right click to handle, in this stage. Is it necessary to set `markEventHandled`?

Reply:
Only `markEventHandled` if you need `isEventHandled`, otherwise it's not needed ;)